src/finufft.cpp

```diff
@@ -542,6 +544,9 @@ int FINUFFT_MAKEPLAN(int type, int dim, BIGINT* n_modes, int iflag,
     return ERR_NTRANS_NOTVALID;
   }
 
+  p = new FINUFFT_PLAN_S;
+  *pp = p;
```

Comment:
I have a questions what if *pp is not None, this means user passes a pp double pointer which already having an allocated finufft_plan_s. 
So should we add free(*pp) if *pp is not None?

Reply:
makeplan should overwrite whatever plan was there before, but you're correct this would be a mem leak unless the old struct was destroyed. So, I agree with you.

Reply:
tl;dr: I changed my mind. Really a finufft_destroy(*pp) would be needed. I tried it. Of course it, or free(*pp), segfaults unless the finufft_plan pointed to is NULL. So we would have to have guru users strictly enforce that otherwise they segfault. Instead I prefer we explain makeplan does not remove an old plan; it must be destroyed by the user first. It's unusual they'd be remaking plans in the same object anyway.

Reply:
I think this makes sense. It's not our responsibility to check if the user accidentally passes in an old plan.